icu4c/source/test/intltest/loctest.cpp

```diff
@@ -3341,3 +3393,117 @@ void LocaleTest::TestBug11053UnderlineTimeZone() {
     Locale l8(locale_str.c_str());
     assertTrue((locale_str + " !l8.isBogus()").c_str(), !l8.isBogus());
 }
+
+void LocaleTest::TestUnd() {
```

Comment:
Please also test the C API:
- uloc_getName(id)
- uloc_getLanguage(id)

Reply:
I submitted a follow-up ticket for this:
https://unicode-org.atlassian.net/browse/ICU-20447